huaweicloud/services/dataarts/resource_huaweicloud_dataarts_factory_script.go

```diff
@@ -0,0 +1,323 @@
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"content": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"connection_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"directory": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"database": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"queue_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			// actually not in the GET response
+			"target_status": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			// actually not in the GET response
+			"approvers": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"approver_name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"configuration": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"created_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"auto_acquire_lock": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceScriptCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		httpUrl = "v1/{project_id}/scripts"
+		product = "dataarts-dlf"
+	)
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating DataArts client: %s", err)
+	}
+	createPath := client.Endpoint + httpUrl
+	createPath = strings.ReplaceAll(createPath, "{project_id}", client.ProjectID)
+	createOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"workspace": d.Get("workspace_id").(string)},
+		JSONBody:         utils.RemoveNil(buildCreateOrUpdateScriptBodyParams(d)),
+		OkCodes:          []int{204},
+	}
+
+	_, err = client.Request("POST", createPath, &createOpt)
+	if err != nil {
+		return diag.Errorf("error creating DataArts script: %s", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", d.Get("workspace_id"), d.Get("name").(string)))
+	return resourceScriptRead(ctx, d, meta)
+}
+
+func resourceScriptRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		mErr    *multierror.Error
+		httpUrl = "v1/{project_id}/scripts/{script_name}"
+		product = "dataarts-dlf"
+	)
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating DataArts client: %s", err)
+	}
+
+	split := strings.Split(d.Id(), "/")
+	if len(split) != 2 {
+		return diag.Errorf("error resolving the id: %s", d.Id())
+	}
+	scriptName := split[1]
```

Comment:
Because `name` is required and can not be modified, also it was set during importing, I think we can use d.Get to fetch the value.
The same thing in Update and Delete Funcs.
```suggestion
	scriptName := d.Get("name").(string)
```